runcoms/zprofile

```diff
@@ -66,11 +66,11 @@ fi
 # Temporary Files
 #
 #
-if [[ -z "$TMPDIR" ]]; then
-  export TMPDIR="/tmp/zsh-$UID"
+# Set TMPDIR if the variable is not set or the directory doesn't exist
+if [[ ! "${TMPDIR}" ]]; then
```

Comment:
I'd like to start moving towards the Google Shell Style Guide which [recommends explicit use of -z and -n](https://google.github.io/styleguide/shell.xml?showone=Testing_Strings#Testing_Strings)